src/main/kotlin/no/skatteetaten/aurora/boober/service/RedeployContext.kt

```diff
@@ -1,68 +1,22 @@
-
-    data class ImageInformation(val lastTriggeredImage: String, val imageStreamName: String, val imageStreamTag: String)
+    open fun didImportImage(openShiftResponse: OpenShiftResponse): Boolean = deploymentConfig?.didImportImage(openShiftResponse)
+            ?: false
 
     open fun isDeploymentRequest(): Boolean = imageStream == null && deploymentConfig != null
 
-    open fun verifyResponse(response: OpenShiftResponse): VerificationResult {
-        val body = response.responseBody
-                ?: return VerificationResult(success = false, message = "No response found")
-        val images = body.at("/status/images") as? ArrayNode
-
-        images?.find { it["status"]["status"].textValue()?.toLowerCase().equals("failure") }?.let {
-            return VerificationResult(success = false, message = it["status"]["message"]?.textValue())
-        }
-
-        return VerificationResult(success = true)
-    }
-
-    open fun didImportImage(response: OpenShiftResponse): Boolean {
-        val body = response.responseBody ?: return true
-        val info = findImageInformation() ?: return true
-        if (info.lastTriggeredImage.isBlank()) {
-            return false
-        }
-
-        val tags = body.at("/status/import/status/tags") as ArrayNode
-        tags.find { it["tag"].asText() == info.imageStreamTag }?.let {
-            val allTags = it["items"] as ArrayNode
-            val tag = allTags.first()
-            return tag["dockerImageReference"].asText() != info.lastTriggeredImage
-        }
-
-        return true
-    }
-
-    open fun findImageInformation(): ImageInformation? {
-        val dc = deploymentConfig?.responseBody ?: return null
+    open fun findImageName(): String? = imageStream?.findImageName()
 
-        val triggers = dc.at("/spec/triggers") as ArrayNode
-        return triggers.find { it["type"].asText().toLowerCase() == "imagechange" }?.let {
-            val (isName, tag) = it.at("/imageChangeParams/from/name").asText().split(':')
-            val lastTriggeredImage = it.at("/imageChangeParams/lastTriggeredImage")?.asText() ?: ""
-            ImageInformation(lastTriggeredImage, isName, tag)
-        }
-    }
+    open fun findImageInformation(): ImageInformation? = deploymentConfig?.findImageInformation()
 
-    open fun findImageName(): String? {
-        val imageInformation = findImageInformation()
-        imageInformation?.let { img ->
-            imageStream?.responseBody?.takeIf { it.openshiftName == img.imageStreamName }?.let {
-                val tags = it.at("/spec/tags") as ArrayNode
-                tags.find { it["name"].asText() == img.imageStreamTag }?.let {
-                    return it.at("/from/name").asText()
-                }
-            }
-        }
-        return null
-    }
+    open fun verifyResponse(openShiftResponse: OpenShiftResponse): VerificationResult =
+            imageStream?.verifyResponse() ?: VerificationResult(success = false, message = "No response found")
```

Comment:
Er det ikke ImageStream i VerificationResult som blir verifisert her, og ikke ImageStreamImport?

Reply:
ImageStreamImport benytter samme objekt

Reply:
Det er to forskjellige objekter, men jeg ser at vi kan bruke samme type for ImageStream og ImageStreamImport siden modellen er såpass "grunn".

Reply:
Denne inneholder en feil som @oyvinder har påpekt til meg.
Og ja, RedeployContext er nok på vei ut :)